OpenKH/egsTools.cs

```diff
@@ -315,7 +315,42 @@ private static Hed.Entry AddFile(string inputFolder, string filename, FileStream
                 encryptionSeed = seed.ReadAllBytes();
                 encryptedData = header.CompressedLength > -2 ? EgsEncryption.Encrypt(compressedData, encryptionSeed) : compressedData;
             }
-			
+
+            IMD imd = null;
+            // We want to replace the original file
+            if (File.Exists(completeFilePath))
+            {
+                Console.WriteLine($"Replacing original: {filename}!");
+
+                using var newFileStream = File.OpenRead(completeFilePath);
+                decompressedData = newFileStream.ReadAllBytes();
+
+                if (filename.IndexOf(".imd") > -1) imd = new IMD(decompressedData);
+
+                if (imd != null && !imd.Invalid) header.RemasteredAssetCount = imd.TextureCount;
+
+                var compressedData = decompressedData.ToArray();
+                var compressedDataLenght = originalHeader.CompressedLength;
+
+                // CompressedLenght => -2: no compression and encryption, -1: no compression 
+                if (originalHeader.CompressedLength > -1)
+                {
+                    compressedData = Helpers.CompressData(decompressedData);
+                    compressedDataLenght = compressedData.Length;
+                }
+
+                header.CompressedLength = compressedDataLenght;
+                header.DecompressedLength = decompressedData.Length;
+                // Encrypt and write current file data in the PKG stream
+
+                // The seed used for encryption is the original data header
+                var seed = new MemoryStream();
+                BinaryMapping.WriteObject<EgsHdAsset.Header>(seed, header);
+
+                encryptionSeed = seed.ReadAllBytes();
+                encryptedData = header.CompressedLength > -2 ? EgsEncryption.Encrypt(compressedData, encryptionSeed) : compressedData;
+            }
+
```

Comment:
Unless I'm mistaken, those this area is 99% identical to [this one](https://github.com/AntonioDePau/KHPCPatchManager/pull/11/files#diff-3baa6eec6f9f45525d3bec234c888e617b59fcc13a3d5a779da5ba21ae06305dR284-R317), the only difference is MDLX vs IMD, so the best approach would be to change MDLX to something more generic, like SDasset (that has a Invalid, Offset, and AssetCount properties) that gets defined based on whether the file is MDLX, IMD, or any future compatible format.